File: server/core/src/main/java/com/musicshop/dto/response/ProductResponse.java
package com.musicshop.dto.response;

import lombok.Builder;

import java.util.Map;
import java.util.UUID;

@Builder
public record ProductResponse(UUID id, String name, int price, String description, String color, String manufacturerName,
                              String imgRef, Map<String, String> characteristics) {
}

File: frontend/src/dto/ProductResponse.ts
import { z } from 'zod';

export const ProductResponseSchema = z.object({
  id: z.string(),
  name: z.string(),
  price: z.number(),
  description: z.string(),
  color: z.string(),
  manufacturerName: z.string(),
  imgRef: z.string(),
  characteristics: z.record(z.string()),
});

export type ProductResponse = z.infer<typeof ProductResponseSchema>;
